### DISCUSSION TITLE
release1.2.1无法正常运行

### DISCUSSION THREAD

**Discussion by Peter-Suchang** [hjenryin/BCJH-Metropolis#217] [Q&A] [answered]
请教一下，不知道还有没有其他人遇到同样的问题，应该是跟以前版本一样的配置和使用方法，但是开始运行之后既不显示百分比进度条，也不输出任何结果，直接显示“按任意键关闭”
<img width="867" alt="image" src="https://github.com/user-attachments/assets/4a24bc40-7461-4500-97af-9b7c54e7538a">
release1.2.0可以正常运行，不过前段时间发现好像达到400W的进程会自动提前结束？请问这个阈值可以像迭代次数那样自行调整的吗？没有找到地方（因为想再手动调整省一些食材，所以想从最高分开始）

**Comment by Peter-Suchang** [comment]
![image](https://github.com/user-attachments/assets/4f4374ed-757a-42c0-819d-1a79e8e6dd0e)
文件应该是齐全的吧 我看跟release1.2.0的相比只少一个states.bin，那个应该是运行之后才生成用来保存最高分避免算出结果不如上次的

**Reply by hjenryin** [reply]
对，应该是全的，states.bin是我不小心传上去的。我国庆来看看

**Reply by Peter-Suchang** [reply]
谢谢大佬，辛苦您啦！states.bin您好像没有误传，1.2.0里面有是因为我自己用那个版本跑过生成的

**Reply by hjenryin** [reply]
抱歉过了这么久，最近实在有点忙。我是了挺久没能复现出这个情况，能试试看把[bcjh.zip](https://github.com/user-attachments/files/17323099/bcjh.zip)里面的exe替换原来的exe，看看结果会有变化吗？如果还是没有，麻烦把出现问题的程序和相关json、csv文件打个包，我在我这里运行一下看看问题是什么。

400W是原来程序里面写死的数字，我下个版本改掉吧。

**Reply by Peter-Suchang** [reply]
哪有哪有，大佬为爱发电感谢还来不及呢！我找时间按您说的试试看再来反馈

**Reply by Peter-Suchang** [reply]
大佬好，我试了下1.2.0的.exe丢进去替换掉1.2.1的.exe可以正常运行的，意思是1.2.1里面的.json跟.csv文件应该是没有问题的吧？我之前删了重下过1.2.1还是上面的问题，没进度也没结果直接结束了

**Reply by Peter-Suchang** [reply]
<img width="867" alt="image" src="https://github.com/user-attachments/assets/3932adf7-f7f1-42ad-bba6-5661f9d8f36b">
好像也没有那么对，这次是部分进程卡在0%或者中间，就已经输出结果结束了

**Reply by hjenryin** [reply]
> 这次是部分进程卡在0%或者中间，就已经输出结果结束了

挺奇怪的，我本来怀疑是中间遇到了一个恶性bug所以直接退出了，我新上传的这个就只是判断一下如果遇到这个bug就把相关信息打印出来。没想到居然能跑完。
中间0%应该只显示问题，因为多线程不能同时写进度条否则会乱，所以当一个线程想更新的时候，发现有人在写，他就下次更新的时候再一起写上。0%结束的就说明他等到结束了都没碰上没人写进度条的情况。
> 意思是1.2.1里面的.json跟.csv文件应该是没有问题的吧

对，主要是现在我拿我自己的账号数据在1.2.1上是好的，所以我不好判断问题出在哪里（可能是某个特定厨师菜谱？），想拿您的出问题的数据试一下。

**Reply by Peter-Suchang** [reply]
不好意思大佬，最近在挪窝，之前把电脑拆了都还没重新装上，可能还得过一小段时间再来请教麻烦您

**Comment by hjenryin** [comment]
哈哈没事不着急，你可以先拿之前的版本用着，把最新版改掉的出问题的厨师暂时禁用，应该结果也不会有太大影响。

________________________________

**Reply by Peter-Suchang** [reply]
大佬好久不见，刚刚出了半个月差回来，台式机还没复原，在笔记本上试了下，还是一样的问题，所以应该不是设备问题，然后删掉toolEquipped.csv也是一样的问题，感觉问题还是在userData.json上面
[userData.json](https://github.com/user-attachments/files/18221194/userData.json)
如果您有空的时候能帮忙看看的话，实在是感激不尽，谢谢谢谢！！！

**Reply by Peter-Suchang** [reply]
而且笔记本上试了下1.2.0也用不了了 跟上面一样的问题T_T

**Comment by hjenryin** [comment]
好奇怪哦，所以1.2.0在有的电脑上可以，有的不行是吗？那这个感觉很难办诶……之前有版本是能成功的吗？我这两天可能会出新版本1.3.0（如果动作快的话也会有1.4.0），1.3.0改的是数据读写部分，主要是为了适配新池，所以可能对这个问题帮助不是很大；1.4.0计划可能会适配游戏里面的数据接口，我们可以到时候看看能不能成功。我也在我这里试一下你的数据，看看能不能运行。如果可以的话，可能是适配电脑的问题。如果真是适配问题，我只能试试看把中间细节通通打印出来，让你运行看看停在哪里，希望能排查到问题。除此之外我就没有什么很好的办法了……On Dec 26, 2024 22:06, Peter-Suchang ***@***.***> wrote:
而且笔记本上试了下1.2.0也用不了了 跟上面一样的问题T_T

**Reply by Peter-Suchang** [reply, marked as answer]
大佬好，好久不见！虽然不太懂为什么，我想应该是您那边做了些努力，最新的1.3.1我可以正常运行出结果了！目前就两个无关紧要的小问题了，一是之前说过的400W自动停止的事，二是新版本进度条显示跟以前不太一样，只要有进程的进度条涨了，所有进度条都会重新打印一次。不过这都不重要了，重要的是可以正常跑了！感谢感谢！

**Reply by hjenryin** [reply]
新年好哇！抱歉先前错过邮件提示没注意到，我其实没找到具体问题是什么，但是我在debug的时候发现了一个相似的问题，正好趁着网页版更新把相关功能都重做了一下，也许是重做之后问题就消失了。要是之后又问题，我再看看;-)

400W自动停止我改了，现在右键编辑`点此运行.bat`，搜索`--target`，就可以改这个分数线了。

进度条按理说是会回到最开头重新打印，把之前的一遍盖住的，所以应该始终只有那几行。是不是你的窗口太小了/字太大了呀？我确实见过，如果窗口不能完整显示整个进度条的话，确实会出现不对的情况。如果是这样的话可以`Ctrl`+`-`缩小字体。如果不是这个原因，那我也不太清楚咋弄欸。

**Reply by Peter-Suchang** [reply]
新年好新年好！！！最新版本以上所有提及的问题都已经没有了，非常感谢！
